core/src/main/java/org/elasticsearch/search/Scroll.java

```diff
@@ -38,18 +38,28 @@
     private final TimeValue keepAlive;
 
     public Scroll(StreamInput in) throws IOException {
-        this.keepAlive = new TimeValue(in);
-    }
+        TimeValue keepAlive = new TimeValue(in);
+        Objects.requireNonNull(keepAlive,"keepAlive must not be null");
+        if(keepAlive.seconds() > 300){
+            throw new IllegalArgumentException("Keep Alive values are restricted to 5 minutes or less.");
+        }
+        this.keepAlive = keepAlive;
+}
 
     /**
      * Constructs a new scroll of the provided keep alive.
      */
     public Scroll(TimeValue keepAlive) {
-        this.keepAlive = Objects.requireNonNull(keepAlive, "keepAlive must not be null");
+        Objects.requireNonNull(keepAlive,"keepAlive must not be null");
+
+        if(keepAlive.seconds() > 300){
```

Comment:
Consider making this `if` statement a function for better consistency when checking for timeouts.